src/service/file_upload_service.rs

```diff
@@ -0,0 +1,336 @@
+        // sanitize and randomize file name and write file into the temporary folder
+        for file in files {
+            let file_name = util::file::generate_unique_filename(
+                &util::file::sanitize_filename(
+                    &file
+                        .name()
+                        .ok_or(Error::Validation(String::from("Invalid file name")))?,
+                ),
+                file.extension(),
+            );
+            let read_file = file.get_contents().await.map_err(persistence::Error::Io)?;
+            self.file_upload_store
+                .write_temp_upload_file(&file_upload_id, &file_name, &read_file)
+                .await?;
+        }
+        Ok(UploadFilesResponse { file_upload_id })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use persistence::file_upload::MockFileUploadStoreApi;
+    use std::sync::Arc;
+    use util::file::MockUploadFileHandler;
+
+    fn get_service(mock_storage: MockFileUploadStoreApi) -> FileUploadService {
+        FileUploadService::new(Arc::new(mock_storage))
+    }
+
+    #[tokio::test]
+    async fn upload_files_baseline() {
+        let file_bytes = String::from("hello world").as_bytes().to_vec();
+        let mut storage = MockFileUploadStoreApi::new();
+        storage
+            .expect_write_temp_upload_file()
+            .returning(|_, _, _| Ok(()));
+        storage
+            .expect_create_temp_upload_folder()
+            .returning(|_| Ok(()));
+        let mut file = MockUploadFileHandler::new();
+        file.expect_name()
+            .returning(|| Some(String::from("invoice")));
+        file.expect_extension()
+            .returning(|| Some(String::from("pdf")));
+        file.expect_get_contents()
+            .returning(move || Ok(file_bytes.clone()));
+        let service = get_service(storage);
+
+        let res = service.upload_files(vec![&file]).await;
+        assert!(res.is_ok());
+        assert_eq!(
+            res.unwrap().file_upload_id,
+            "00000000-0000-0000-0000-000000000000".to_owned()
+        );
+    }
+
+    #[tokio::test]
+    async fn upload_files_baseline_fails_on_folder_creation() {
+        let file_bytes = String::from("hello world").as_bytes().to_vec();
+        let mut storage = MockFileUploadStoreApi::new();
+        storage.expect_create_temp_upload_folder().returning(|_| {
+            Err(persistence::Error::Io(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "test error",
+            )))
+        });
+        let mut file = MockUploadFileHandler::new();
+        file.expect_name()
+            .returning(|| Some(String::from("invoice")));
+        file.expect_extension()
+            .returning(|| Some(String::from("pdf")));
+        file.expect_get_contents()
+            .returning(move || Ok(file_bytes.clone()));
+        let service = get_service(storage);
+
+        let res = service.upload_files(vec![&file]).await;
+        assert!(res.is_err());
```

Comment:
can we start to use assertion macros (assert!, assert_eq!) for validation, like in our guide?